triton/apps/vscode.rst

```diff
@@ -0,0 +1,64 @@
+VSCode on Triton
+================
+
+VSCode is available through :doc:`../usage/ood`.  If you want to
+connect through your own computer, read below.
+
+"Remote SSH" is a nice way to work on a remote computer, but on
+Triton, that will run everything on the login node.  For light use
+this is OK, but for heavy calculations, since it will affect other
+login node usage.
+
+
+VSCode remote SSH host directly to interactive job
+--------------------------------------------------
+
+Sometimes you want more resources than the login node.  This section
+presents a way to have VSCode directly connect to a job resource
+allocation on Triton - so you can do larger calculations / use more
+memory / etc. without interfering with others.  **Note that for real
+production calculations,** you should use :doc:`../tut/serial`, and
+*not* run stuff through your editor, **since everything gets lost when
+your connection dies.**
+
+This section contains original research and may not fully work, and
+**may only work on Linux/Mac right now**.
+
+In you ``~/.ssh/config``, add this block to define a server
+``triton-vscode``.  For more information ``.ssh/config``, including
+what these mean and what else you might need in here, see
+:doc:`/scicomp/ssh`::
+
+  Host triton-vscode
+      ProxyCommand ssh triton /share/apps/ssh-node-proxycommand --partition=interactive --time=1:00:00
+      StrictHostKeyChecking no
+      UserKnownHostsFile /dev/null
+      User darstr1
+
+  # You also need a triton alias here:
+
+  Host triton
+      HostName triton.aalto.fi
+      # ... any other thing you need for connecting to triton.
+
+
+Now, with VSCode's Remote SSH, you can select the ``triton-vscode``
+remote.  It will ssh to Triton, request a job, and then directly
+connect to the job.  **Configure the job requirements in the
+ProxyCommand line** (you could make different lines).
```

Comment:
For more details have a look at: :doc:`/triton/ref/#job-submission`
(Not entirely sure whether this is the right link.

Reply:
Added via other commit where I also gave anchors to the quick ref.